src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -426,6 +431,67 @@ def create(self, daemon_spec: CephadmDaemonSpec) -> str:
 
         return self.mgr._create_daemon(daemon_spec)
 
+    def create_realm_zonegroup_zone(self, spec: RGWSpec, rgw_id: str):
+        if utils.get_cluster_health(self.mgr) != 'HEALTH_OK':
+            raise OrchestratorError('Health not ok, will try agin when health ok')
+
+        #get keyring needed to run rados commands
+        ret, keyring, err = self.mgr.check_mon_command({
+            'prefix': 'auth get-or-create',
+            'entity': f"{utils.name_to_config_section('rgw')}.{rgw_id}",
+            'caps': ['mon', 'allow *',
+                     'mgr', 'allow rw',
+                     'osd', 'allow rwx'],
+        })
```

Comment:
the logic is duplicated by both `create()` and `create_realm_zonegroup_zone` ..  create a dedicated function in the RGW class for getting the keyring?